boards/common/stm32/include/l4/cfg_clock_80_1.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *               2019 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_common_stm32
+ * @{
+ *
+ * @file
+ * @brief       Configure STM32L4 clock using 80MHz core clock and LSE (32.768kHz)
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ */
+
+#ifndef L4_CFG_CLOCK_80_1_H
+#define L4_CFG_CLOCK_80_1_H
+
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Clock system configuration
+ * @{
+ */
+/* 0: no external high speed crystal available
+ * else: actual crystal frequency [in Hz] */
+#define CLOCK_HSE           (0)
+
+#ifndef CLOCK_LSE
+/* 0: no external low speed crystal available,
+ * 1: external crystal available (always 32.768kHz)
+ */
+#define CLOCK_LSE           (1)
+#endif
+
+/* 0: enable MSI only if HSE isn't available
+ * 1: always enable MSI (e.g. if USB or RNG is used)*/
+#define CLOCK_MSI_ENABLE    (1)
```

Comment:
What's the logic on which parameters to make configurable (`#ifndef`) and which not?

Reply:
From reading the code, it allows for configuring how the different clocks are configured using CFLAGS, knowing that there are good defaults (LSE + PLL). Here one could select LSI or LSE (default) or not use the PLL.
That might be good candidates for the Kconfig migration :)
I would also have the PLL_M/N/R values configurable if the PLL is used. That would allow finer selection of the core clock speed.

Do you want me to remove the ifndef ?

Reply:
No, I was thinking more of: if this header file is meant to be a default or boilerplate for boards, would it make sense to define all of them with `#ifndef`?

Reply:
I dropped the `#ifndef`. I think we should have a more general discussion on how to handle clock configuration consistently across boards. Hope this is fine :)